src/sentry/services/hybrid_cloud/app/__init__.py

```diff
@@ -1,58 +1,120 @@
 from __future__ import annotations
 
 import abc
+import datetime
+import hmac
 from dataclasses import dataclass, field
-from typing import TYPE_CHECKING, List
+from hashlib import sha256
+from typing import Any, Dict, List, Optional, TypedDict
 
 from sentry.constants import SentryAppInstallationStatus
+from sentry.db.models.fields.jsonfield import JSONField
 from sentry.services.hybrid_cloud import InterfaceWithLifecycle, silo_mode_delegation, stubbed
+from sentry.services.hybrid_cloud.filter_query import FilterQueryInterface
 from sentry.silo import SiloMode
 
-if TYPE_CHECKING:
-    from sentry.models import SentryApp, SentryAppInstallation
 
+class SentryAppInstallationFilterArgs(TypedDict, total=False):
+    id: int
+    organization_id: int
+    uuid: str
+    date_deleted: Optional[datetime.datetime]
+    status: int  # SentryAppInstallationStatus
 
-class AppService(InterfaceWithLifecycle):
+
+@dataclass
+class ApiSentryAppInstallation:
+    id: int = -1
+    organization_id: int = -1
+    status: int = SentryAppInstallationStatus.PENDING
+    uuid: str = ""
+    sentry_app: ApiSentryApp = field(default_factory=lambda: ApiSentryApp())
+
+
+@dataclass
+class ApiApiApplication:
```

Comment:
I'll take a stab at renaming our api classes here  in a separate PR after we validate this works.